torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -23,8 +24,15 @@ DistAutogradContainer::DistAutogradContainer()
       worker_id_(0),
       initialized_(false),
       next_autograd_message_id_(0),
-      max_id_(0) {}
+      max_id_(0) {
+  terminateWatchdog_.store(false);
+  cleanupWatchdogThread_ = std::thread(&DistAutogradContainer::cleanupContextWatchdog, this);
+}
 
+DistAutogradContainer::~DistAutogradContainer() {
+  terminateWatchdog_.store(true);
+  cleanupWatchdogThread_.join();
```

Comment:
Hmm, so if we immediately create and then destroy a `DistAutogradContainer`, am I correct in thinking that it the destruction will always take at least `kContextTimeout` to complete? Can we use a condition variable instead?